ElementX/Sources/Screens/HomeScreen/View/HomeScreen.swift

```diff
@@ -87,10 +93,96 @@ struct HomeScreen: View {
         .toolbar { toolbar }
         .background(Color.compound.bgCanvasDefault.ignoresSafeArea())
         .track(screen: .home)
+        .introspect(.viewController, on: .iOS(.v16, .v17)) { controller in
+            Task {
+                if bloomView == nil {
+                    setBloomView(controller: controller)
+                }
+            }
+            let isHidden = controller.navigationController?.topViewController != controller || isSearching
+            bloomView?.isHidden = isHidden
+            gradientView?.isHidden = isHidden
+            navigationBarContainer?.clipsToBounds = !isHidden
+            hairlineView?.isHidden = isHidden || !scrollViewAdapter.scrollViewIsAtTopEdgePublisher.value
+            if !isHidden {
+                updateBloomCenter()
+            }
+        }
+        .onReceive(scrollViewAdapter.scrollViewIsAtTopEdgePublisher.removeDuplicates()) { value in
+            hairlineView?.isHidden = !value
+            guard let gradientView else {
+                return
+            }
+            if value {
+                UIView.transition(with: gradientView, duration: 0.3, options: .curveEaseIn) {
+                    gradientView.alpha = 0
+                }
+            } else {
+                gradientView.alpha = 1
+            }
+        }
     }
     
     // MARK: - Private
     
+    private var bloomGradient: some View {
+        LinearGradient(colors: [.clear, .compound.bgCanvasDefault], startPoint: .top, endPoint: .bottom)
+            .mask {
+                LinearGradient(stops:
+                    [.init(color: .white, location: 0.75), .init(color: .clear, location: 1.0)],
+                    startPoint: .leading, endPoint: .trailing)
```

Comment:
Should probably be like this (I thought SwiftFormat would be upset in some way about this).
```suggestion
                LinearGradient(stops: [.init(color: .white, location: 0.75), .init(color: .clear, location: 1.0)],
                               startPoint: .leading,
                               endPoint: .trailing)
```